sql/src/main/java/org/apache/druid/sql/calcite/rule/DruidJoinRule.java

```diff
@@ -97,18 +98,19 @@ public void onMatch(RelOptRuleCall call)
 
     final DruidRel<?> newLeft;
     final DruidRel<?> newRight;
+    final Filter leftFilter;
     final List<RexNode> newProjectExprs = new ArrayList<>();
 
     // Already verified to be present in "matches", so just call "get".
     // Can't be final, because we're going to reassign it up to a couple of times.
     ConditionAnalysis conditionAnalysis = analyzeCondition(join.getCondition(), join.getLeft().getRowType()).get();
 
-    if (left.getPartialDruidQuery().stage() == PartialDruidQuery.Stage.SELECT_PROJECT
-        && left.getPartialDruidQuery().getWhereFilter() == null) {
+    if (left.getPartialDruidQuery().stage() == PartialDruidQuery.Stage.SELECT_PROJECT) {
```

Comment:
This may not work right if `left` is a DruidJoinQueryRel or DruidOuterQueryRel instead of a DruidQueryRel (since `leftFilter` is only supported for regular tables).

Reply:
oops. good catch. will fix this.